.github/workflows/integrate.yaml

```diff
@@ -0,0 +1,104 @@
+name: CI
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  lint:
+    name: Lint Check
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: sudo apt-get install python3-pip tox
+
+    - name: Lint code
+      run: tox -e lint
+
+  unit:
+    name: Unit Test
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: sudo apt-get install python3-pip tox
+
+    - name: Run unit tests
+      run: tox -e unit
+
+  deploy:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repo
+      uses: actions/checkout@v2
+
+    - uses: balchua/microk8s-actions@v0.2.2
+      with:
+        addons: '["dns", "storage", "rbac"]'
+
+    # Remove once this is fixed:
+    # https://github.com/actions/virtual-environments/issues/3185
+    - name: Twiddle with MicroK8s
+      run: |
+        set -eux
+        echo '--kubelet-preferred-address-types="InternalIP,Hostname,InternalDNS,ExternalDNS,ExternalIP"' | sudo tee -a /var/snap/microk8s/current/args/kube-apiserver
+        sudo snap restart microk8s
+        sg microk8s -c 'microk8s status --wait-ready'
+        kubectl -nkube-system rollout status ds/calico-node
+        kubectl wait --for=condition=available -nkube-system deployment/coredns deployment/hostpath-provisioner
+
+    - name: Install dependencies
+      run: |
+        set -eux
+        sudo pip3 install charmcraft
+        sudo snap install juju --classic
+        sudo snap install juju-helpers --classic
+        sudo snap install juju-wait --classic
+        sudo snap install yq
+
+    - name: Bootstrap Juju
+      run: |
+        set -eux
+        sg microk8s -c 'juju bootstrap microk8s uk8s'
+        juju add-model ci-test
+
+    - name: Deploy charm
+      run: |
+        set -eux
+        charmcraft build
+        juju bundle deploy -b tests/bundle.yaml --build
+        juju wait -wvt 300
+        juju status
+
+    - name: Test charm
+      run: |
+        set -eux
+        DASHBOARD_URL=`kubectl -n ci-test get svc kubeflow-dashboard -o jsonpath='{.spec.clusterIP}'`
+        curl DASHBOARD_URL:8082
+        exit $?
```

Comment:
I don't think this will do anything, because we run `set -eux` above, so if `curl` fails at all, the script will abort there. We could throw in the `-f` flag to curl to make it fail if it gets anything other than a 200